rust/src/delta.rs

```diff
@@ -561,6 +569,34 @@ impl DeltaTable {
         }
     }
 
+    pub fn get_files_by_partitions(
+        &self,
+        filters: Vec<PartitionFilter<&str>>,
+    ) -> Result<Vec<String>, DeltaTableError> {
+        let partitions_number = match &self.state.current_metadata {
+            Some(metadata) if !metadata.partition_columns.is_empty() => {
+                metadata.partition_columns.len()
+            }
+            _ => return Err(DeltaTableError::LoadPartitions),
+        };
+        let files = self
+            .state
+            .files
+            .iter()
+            .filter(|f| {
+                f.splitn(partitions_number + 1, self.storage.get_separator())
```

Comment:
nitpick, should call `self.storage.get_separator()` outside of the loop once.

Reply:
Yep right!

I changed this by using the ` /` separator because it works in all cases except one.
The only case that could raise an error is a  `Delta Table` partitioned **created** on a Windows machine but I don't know how to test/detect it. A `DeltaTable` could be created in S3 and read on a local Windows machine...